### PR TITLE
Make InheritingDictionary override values.

### PR DESCRIPTION
**Pull request by jamadden**

### CODE REVIEW

File: src/zope/i18n/locales/inheritance.py [hunk 1]
```diff
@@ -179,10 +179,9 @@ class InheritingDictionary(Inheritance, dict):
       >>> list(locale.data.items())
       [(1, 'eins'), (2, 'two'), (3, 'three')]
 
-    We currently fail to override ``values``, but instead inherited
-    data can be seen in the ``value`` method::
+    We also override ``values``::
 
-      >>> sorted(locale.data.value())
+      >>> sorted(locale.data.values())
       ['eins', 'three', 'two']
```
Comment (mgedmin) [inline src/zope/i18n/locales/inheritance.py]: Let's not sort, maybe -- the values should be returned in the same order as `keys()` and `items()`, and the tests for both of those do not call `sorted()`.

File: src/zope/i18n/locales/inheritance.py [hunk 2]
```diff
@@ -226,6 +225,8 @@ def items(self):
     def keys(self):
         return list(self._make_reified_inherited_dict().keys())
 
-    def value(self):
-        # XXX: Was this meant to override values() and is just a typo?
+    def values(self):
         return list(self._make_reified_inherited_dict().values())
+
+    # Preserve the old name for compatibility
+    value = values
```
Comment (mgedmin) [inline src/zope/i18n/locales/inheritance.py]: Is it hard to add a deprecation warning on the use of this?

File: src/zope/i18n/locales/inheritance.py (hunk 1 above)
Comment (jamadden) [reply src/zope/i18n/locales/inheritance.py]: Or maybe those others should call sorted? They are all the result of iterating across views of a dict, and dicts are inherently unordered (prior to 3.7, where the language spec has changed). The fact that the tests pass at all is an implementation artifact (hashes of ints are the int value, dicts iterate in hash order).

File: src/zope/i18n/locales/inheritance.py (hunk 2 above)
Comment (jamadden) [reply src/zope/i18n/locales/inheritance.py]: We don't have a dependency on zope.deprecation, so we would either need to add one or write the code manually. That would have to be tested, which either means in the docstring (thus documenting it in perpetuity) or a whole new test file.  That all seemed like overkill to me.

Maybe we should just drop the old name entirely, on the theory that no one was actually using it?

Comment (jamadden) [reply src/zope/i18n/locales/inheritance.py]: `zope.deprecation` is a lightweight dependency, so I added that.

**Comment by jamadden** [comment]
I think I've addressed the feedback (which, as always, was very valuable; thank you!). Once Travis goes green, I plan to merge and then release if there are no objections.

**Pull request merged**